src/multivariate/dirichlet.jl

```diff
@@ -180,7 +180,7 @@ function _rand!(d::Union{Dirichlet,DirichletCanon}, x::AbstractVector{T}) where
     n = length(x)
     α = d.alpha
     for i in 1:n
```

Comment:
`for i in eachindex(x)` would be faster and more concise

Reply:
true, but i'm trying not to change too much non-constructor code as part of this PR.

Care to make a fix as a new PR?